### PR TITLE
Close connection java wrapper

### PR DESCRIPTION
**Pull request by dsquirek**

### CODE REVIEW

File: ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java [hunk 1]
```diff
@@ -28,6 +29,10 @@ protected void processHTTPResponse(HttpResponse httpresp)
 		}
 
 	}
+	
+	protected void closeConnection(HttpResponse response) throws IOException{
+		EntityUtils.consume(response.getEntity());
```
Comment (thomasmahoney) [inline ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java]: where are you actually calling this? also, what about the connections in .NET?

File: ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java [hunk 2]
```diff
@@ -28,6 +29,10 @@ protected void processHTTPResponse(HttpResponse httpresp)
 		}
 
 	}
+	
+	protected void closeConnection(HttpResponse response) throws IOException{
```
Comment (thefroatgt) [inline ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java]: should this be called somewhere?

File: ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java (hunk 1 above)
Comment (dsquirek) [inline ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java]: from what I've seen about restsharp connections are automatically closed after the request is executed.

File: ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java (hunk 2 above)
Comment (dsquirek) [inline ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java]: yes it should, I have now made the change to correct this.

File: ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ServiceBase.java [hunk 3]
```diff
@@ -29,6 +29,8 @@ protected Response makeAPICall(Request parameters, OAuthCredentials auth,
 			Response resp = type.newInstance();
 
 			resp.processHTTPResponse(httpresp);
+			
+			resp.closeConnection(httpresp);
```
Comment (thefroatgt) [inline ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ServiceBase.java]: this should be in a finally block, along with appropriate tests to make sure we don't throw a null ref out of the finaly block

**Pull request merged**